alpha/apps/kaltura/lib/kCategoryEventHandler.php

```diff
@@ -0,0 +1,305 @@
+	 * @see kObjectChangedEventConsumer::shouldConsumeChangedEvent()
+	 */
+	public function shouldConsumeChangedEvent(BaseObject $object, array $modifiedColumns) {
+		if ($object instanceof categoryEntry && in_array(categoryEntryPeer::STATUS, $modifiedColumns)
+			&& $object->getStatus() == CategoryEntryStatus::ACTIVE
+			&& $object->getColumnsOldValue(categoryEntryPeer::STATUS) == CategoryEntryStatus::PENDING)
+		{
+			return true;
+		}
+		
+		if ($object instanceof category)
+		{
+			$oldCustomDataValues = $object->getCustomDataOldValues();
+			$oldAggregationChannels = isset ($oldCustomDataValues[''][category::AGGREGATION_CATEGORIES]) ? $oldCustomDataValues[''][category::AGGREGATION_CATEGORIES] : '';
+			
+			if ($oldAggregationChannels != $object->getAggregationCategories())
+			{
+				return true;
+			}
+		}
+		
+		return false;
+	}
+
+	/* (non-PHPdoc)
+	 * @see kObjectCreatedEventConsumer::objectCreated()
+	 */
+	public function objectCreated(BaseObject $object) {
+		if ($object instanceof categoryEntry)
+		{
+			$this->handleCategoryEntryCreated($object);
+		}
+		
+		if ($object instanceof category)
+		{
+			$this->handleCategoryCreated ($object);
+		}
+		
+		return true;
+		
+	}
+	
+	protected function handleCategoryCreated (category $object)
+	{
+		if (!$object->getAggregationCategories())
+		{
+			KalturaLog::info ("Category [" . $object->getId() . "] has no aggregation channels" );
+			return true;
+		}
+		
+		$this->addToAggregationCategories($object, explode (',', $object->getAggregationCategories()));
+	}
+	
+	protected function addCopyJobToAggregationChannel (category $object, $aggregationCategoryId)
+	{
+		$templateObject = new categoryEntry();
+		$templateObject->setCategoryId($aggregationCategoryId);
+		
+		$filter = new categoryEntryFilter();
+		$filter->set("_eq_category_id", $object->getId());
+		kJobsManager::addCopyJob($object->getPartnerId(), CopyObjectType::CATEGORY_ENTRY, $filter, $templateObject);
+	}
+	
+	protected function addToAggregationCategories (category $object, array $aggregationCatIds)
+	{
+		$aggregationCategories = categoryPeer::retrieveByPKs($aggregationCatIds);
+		foreach ($aggregationCategories as $aggregationCategory)
+		{
+			/* @var $aggregationCategory category */
+			$currentPublishingCategories = explode (',', $aggregationCategory->getPublishingCategories());
+			$currentPublishingCategories[] = $object->getId();
+			$currentPublishingCategories = array_unique($currentPublishingCategories);
+			$aggregationCategory->setPublishingCategories(implode(',', $currentPublishingCategories));
+			$aggregationCategory->save();
+			
+			$this->addCopyJobToAggregationChannel ($object, $aggregationCategory->getId());
+		}
+	}
+	
+	protected function handleCategoryEntryCreated (categoryEntry $object)
+	{
+			$category = categoryPeer::retrieveByPK($object->getCategoryId());
+			if (!$category)
+			{
+				KalturaLog::info("category [" . $object->getCategoryId() . "] does not exist in the system.");
+				return true;
```

Comment:
No need to return true if this method is not expected to return a boolean value. It can just be `return;`